/* comment */
// comment

syntax = "proto3";

service TypesDemoService {
    rpc TypesDemo (TestRequest) returns (TestResponse) {}
}

message TestRequest {
    string str_field = 1;
    int32 int_field = 2;
    bool bool_field = 3;
    float float_field = 4;
    enum EnumField { VAL1=0; VAL2=1; }
    EnumField enum_field = 5;
    repeated string repeated_str_field = 6;
    map<string, int32> map_field = 7;
    Number number_field = 8;
}

message TestResponse {
    string str_field = 1;
    int32 int_field = 2;
    bool bool_field = 3;
    float float_field = 4;
    enum EnumField { VAL1=0; VAL2=1; }
    EnumField enum_field = 5;
    repeated string repeated_str_field = 6;
    map<string, int32> map_field = 7;
    Number number_field = 8;
}

message Number {
    float value = 1;
}
